CHANGELOG.unreleased.md

```diff
@@ -44,6 +44,50 @@ const link = <a href="example.com">http://example.com</a>;
 
 -->
 
+#### Handlebars: Fix handling of whitespace and line breaks ([#6354] by [@chadian])
+
+This fixes a variety of whitespace and line break usecases within handlebars and Glimmer templates.
+
+<!-- prettier-ignore -->
+```hbs
+// Input
+<SomeComponent />{{name}}
+
+Some sentence with  {{dynamic}}  expressions.
+
+
+
+sometimes{{nogaps}}areimportant<Hello></Hello>
+{{name}}  is your name
+
+// Output (Prettier stable)
+<SomeComponent />
+{{name}}
+Some sentence with
+{{dynamic}}
+expressions.
+
+
+
+sometimes
+{{nogaps}}
+areimportant
+<Hello />
+{{name}}
+is your name
+
+// Output (Prettier master)
+<SomeComponent />{{name}}
+
+Some sentence with {{dynamic}} expressions.
+
+
+
+sometimes{{nogaps}}areimportant
+<Hello />
```

Comment:
does this introduce unwanted whitespace between `areimportant` and `<Hello>`?

Reply:
Hmm, I'm not sure what's best here. Wouldn't this usually be the desired result? If that's the case once a prettier-ignore is added I would use that to preserve the line.

Reply:
Yeah, I think it's true that it would usually be the desired result, and that wanting to ensure that there is no whitespace between the text and the component is an edge case.

In my mind though, specifying `sometext<SomeComponent />` is a strong hint to prettier that the lack of whitespace is important here. That's also true of `sometimes{{nogaps}}areimportant`. 

Perhaps we should move this to an issue to gather some more thoughts and land this PR?

Reply:
Sounds good to me! Thanks for reviewing and looking at everything so quickly.

Reply:
I'll create that issue once this lands so the issue is pointed against the current playground for reference.